packages/office-addin-dev-settings/src/sideload.ts

```diff
@@ -79,18 +80,24 @@ export async function generateSideloadFile(app: OfficeApp, manifest: ManifestInf
     if (!extEntry) {
       throw new ExpectedError("webextension was not found.");
     }
-
+    
     const webExtensionXml = templateZip
-      .readAsText(extEntry)
-      .replace(/00000000-0000-0000-0000-000000000000/g, manifest.id)
-      .replace(/1.0.0.0/g, manifest.version);
+    .readAsText(extEntry)
+    .replace(/00000000-0000-0000-0000-000000000000/g, manifest.id)
+    .replace(/1.0.0.0/g, manifest.version);
 
     templateZip.getEntries().forEach(function (entry) {
       var data: Buffer = entry.getData();
-      if (entry == extEntry) {
+      if (entry == extEntry && manifest.manifestType == ManifestType.XML) {
         data = Buffer.from(webExtensionXml);
       }
       outZip.addFile(entry.entryName, data, entry.comment, entry.attr);
+      
+      const webExtensionFolderPath = webExtensionPath.replace(/(\/[^\/]*)$/, '');
+      // If manifestType is JSON, remove the web extension folder
+      if (entry.entryName.startsWith(webExtensionFolderPath) && manifest.manifestType == ManifestType.JSON) {
+        outZip.deleteFile(entry.entryName);
+      }
```

Comment:
Hi @millerds , there will show an error if we leave it here. Because of original web-extension, a fake add-in is auto launched with errors. So, we have to remove web extension from the zip.

Reply:
Makes sense (I didn't know iif the host would throw an error).

You should still use the if clause I suggested instead of adding the entry only to delete it later.

Reply:
Still the same error. Have to remove it.

Reply:
I see now . . . it's removing the other file in that directory and not just the data one.  In that case then webExtensionFolderPath should be defined outside of the foreach statement and the if clause be around the outZip.addFile statement with the logic reversed (i.e. add if not json or not folder path.

Reply:
Thanks @millerds , updated